src/test/java/seedu/address/logic/commands/recipe/RecipeRemoveIngredientCommandTest.java

```diff
@@ -8,35 +8,54 @@
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.ModelManager;
+import seedu.address.model.ingredient.IngredientName;
+import seedu.address.model.ingredient.IngredientQuantity;
 import seedu.address.model.ingredient.exceptions.IngredientNotFoundException;
 
 public class RecipeRemoveIngredientCommandTest {
     private static final Index VALID_RECIPE_INDEX = new Index(0);
     private static final Index OUT_OF_BOUNDS_RECIPE_INDEX = new Index(1);
+    private static final IngredientName VALID_INGREDIENT_NAME = ALMOND.getName();
+    private static final Optional<IngredientQuantity> VALID_INGREDIENT_QUANTITY = Optional.of(ALMOND.getQuantity());
 
     @Test
     public void constructor_validInput() {
-        RecipeRemoveIngredientCommand c = new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX, ALMOND);
-        assertEquals(c, new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX, ALMOND));
+        RecipeRemoveIngredientCommand c = new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX,
+                VALID_INGREDIENT_NAME, VALID_INGREDIENT_QUANTITY);
+        assertEquals(c, new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX,
+                VALID_INGREDIENT_NAME, VALID_INGREDIENT_QUANTITY));
     }
 
     @Test
     public void constructor_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX, null));
-        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(null, ALMOND));
-        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX,
+                null, null)); // index, null, null
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(null,
+                null, VALID_INGREDIENT_QUANTITY)); // null, null, quantity
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(null,
+                VALID_INGREDIENT_NAME, null)); // null, name, null
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX,
+                VALID_INGREDIENT_NAME, null)); // index, name, null
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(VALID_RECIPE_INDEX,
+                null, VALID_INGREDIENT_QUANTITY)); // index, null, quantity
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(null,
+                VALID_INGREDIENT_NAME, VALID_INGREDIENT_QUANTITY)); // null, name, quantity
+        assertThrows(NullPointerException.class, () -> new RecipeRemoveIngredientCommand(null, null, null)); // all null
     }
 
     @Test
     public void execute_validInput() throws CommandException {
```

Comment:
The error was not picked up here because somehow the test used the `RecipeAddIngredientCommand` instead of the `RecipeRemoveIngredientCommand`.